test/exportsField.js

```diff
@@ -130,6 +130,183 @@ describe("Process exports field", function exportsField() {
+		},
+		{
+			name: "sample #11",
+			expect: ["./ok.js"],
+			suite: [
+				{
+					"./#foo": "./ok.js",
+					"./module": "./ok.js",
+					"./🎉": "./ok.js",
+					"./%F0%9F%8E%89": "./other.js",
+					"./bar#foo": "./ok.js",
+					"./#zapp/": "./"
+				},
+				"./bar#foo",
+				[]
+			]
+		},
+		{
+			name: "sample #12",
+			expect: ["./ok.js#abc"],
+			suite: [
+				{
+					"./#foo": "./ok.js",
+					"./module": "./ok.js",
+					"./🎉": "./ok.js",
+					"./%F0%9F%8E%89": "./other.js",
+					"./bar#foo": "./ok.js",
+					"./#zapp/": "./"
+				},
+				"./#zapp/ok.js#abc",
+				[]
+			]
+		},
+		{
+			name: "sample #13",
+			expect: ["./ok.js?abc"],
+			suite: [
+				{
+					"./#foo": "./ok.js",
+					"./module": "./ok.js",
+					"./🎉": "./ok.js",
+					"./%F0%9F%8E%89": "./other.js",
+					"./bar#foo": "./ok.js",
+					"./#zapp/": "./"
+				},
+				"./#zapp/ok.js?abc",
+				[]
+			]
+		},
+		{
+			name: "sample #14",
+			expect: ["./🎉.js"],
+			suite: [
+				{
+					"./#foo": "./ok.js",
+					"./module": "./ok.js",
+					"./🎉": "./ok.js",
+					"./%F0%9F%8E%89": "./other.js",
+					"./bar#foo": "./ok.js",
+					"./#zapp/": "./"
+				},
+				"./#zapp/🎉.js",
+				[]
+			]
+		},
+		{
+			name: "sample #15",
+			expect: ["./%F0%9F%8E%89.js"],
+			suite: [
+				{
+					"./#foo": "./ok.js",
+					"./module": "./ok.js",
+					"./🎉": "./ok.js",
+					"./%F0%9F%8E%89": "./other.js",
+					"./bar#foo": "./ok.js",
+					"./#zapp/": "./"
+				},
+				// "🎉" percent encoded
+				"./#zapp/%F0%9F%8E%89.js",
+				[]
+			]
+		},
+		{
+			name: "sample #16",
+			expect: ["./ok.js"],
+			suite: [
+				{
+					"./#foo": "./ok.js",
+					"./module": "./ok.js",
+					"./🎉": "./ok.js",
+					"./%F0%9F%8E%89": "./other.js",
+					"./bar#foo": "./ok.js",
+					"./#zapp/": "./"
+				},
+				"./🎉",
+				[]
+			]
+		},
+		{
+			name: "sample #17",
+			expect: ["./other.js"],
+			suite: [
+				{
+					"./#foo": "./ok.js",
+					"./module": "./ok.js",
+					"./🎉": "./ok.js",
+					"./%F0%9F%8E%89": "./other.js",
+					"./bar#foo": "./ok.js",
+					"./#zapp/": "./"
+				},
+				"./%F0%9F%8E%89",
+				[]
+			]
+		},
+		{
+			name: "sample #18",
+			expect: ["./ok.js"],
+			suite: [
+				{
+					"./#foo": "./ok.js",
+					"./module": "./ok.js",
+					"./🎉": "./ok.js",
+					"./%F0%9F%8E%89": "./other.js",
+					"./bar#foo": "./ok.js",
+					"./#zapp/": "./"
+				},
+				"./module",
+				[]
+			]
+		},
+		{
+			name: "sample #19",
+			expect: [],
+			suite: [
+				{
+					"./#foo": "./ok.js",
+					"./module": "./ok.js",
+					"./🎉": "./ok.js",
+					"./%F0%9F%8E%89": "./other.js",
+					"./bar#foo": "./ok.js",
+					"./#zapp/": "./"
+				},
+				"./module#foo",
+				[]
+			]
+		},
+		{
+			name: "sample #20",
+			expect: [],
+			suite: [
+				{
+					"./#foo": "./ok.js",
+					"./module": "./ok.js",
+					"./🎉": "./ok.js",
+					"./%F0%9F%8E%89": "./other.js",
+					"./bar#foo": "./ok.js",
+					"./#zapp/": "./"
+				},
+				"./module?foo",
+				[]
+			]
+		},
 		//#endregion
```

Comment:
add a test case for `{ "./a?b?c/": "./" }` `"./a?b?c/d?e?f"` -> `./d?e?f`

Looks weird I know^^